src/client/datascience/commands/exportCommands.ts

```diff
@@ -69,26 +70,28 @@ export class ExportCommands implements IDisposable {
 
         if (editor && editor.model) {
             const interpreter = editor.notebook?.getMatchingInterpreter();
-            return this.export(editor.model, undefined, undefined, interpreter);
+            return this.export(editor.model.getContent(), editor.model.file, undefined, undefined, interpreter);
         } else {
             return this.export(undefined, undefined, undefined, undefined);
         }
     }
 
     private async export(
-        model?: INotebookModel,
+        contents?: string,
+        source?: Uri,
         exportMethod?: ExportFormat,
         defaultFileName?: string,
         interpreter?: PythonEnvironment
     ) {
-        if (!model) {
-            // if no model was passed then this was called from the command palette,
+        if (!contents || !source) {
```

Comment:
Can you check calling this Export command from the command palette with all our types of editor? I thought when I was looking at this recently there was a case where a Model was getting passed in. I think that it might have been Old editor active editor = Command is called with all undefined. CustomEditor active editor = Command is called with an INotebookModel. Not 100% sure I'm recalling correctly.

Reply:
Good thing I double checked. Found another issue. Interactive window would show the export commands if a notebook had ever been opened. Since the active editor here is from the notebook, the export command would essentially do nothing.

The bug was in our tracking of active editor. We weren't firing the change of the active editor on close.